src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,168 @@ class spell_pri_power_word_shield : public SpellScriptLoader
     }
 };
 
-// 33110 - Prayer of Mending Heal
-class spell_pri_prayer_of_mending_heal : public SpellScriptLoader
+// Base class used by various prayer of mending spells
+class spell_pri_prayer_of_mending_SpellScriptBase : public SpellScript
+{
+public:
+    bool Load() override
+    {
+        _spellInfoHeal = sSpellMgr->GetSpellInfo(SPELL_PRIEST_PRAYER_OF_MENDING_HEAL, DIFFICULTY_NONE);
+        if (!_spellInfoHeal)
+            return false;
+
+        _healEffectDummy = _spellInfoHeal->GetEffect(EFFECT_0);
+        if (!_healEffectDummy)
+            return false;
+
+        return true;
+    }
+
+    void CastPrayerOfMendingAura(Unit* caster, Unit* target, uint8 stack)
+    {
+        uint32 basePoints = caster->SpellHealingBonusDone(target, _spellInfoHeal, _healEffectDummy->CalcValue(caster), HEAL, _healEffectDummy);
+        CustomSpellValues values;
+        values.AddSpellMod(SPELLVALUE_AURA_STACK, stack);
+        values.AddSpellMod(SPELLVALUE_BASE_POINT0, basePoints);
+        caster->CastCustomSpell(SPELL_PRIEST_PRAYER_OF_MENDING_AURA, values, target, TRIGGERED_FULL_MASK);
+    }
+
+protected:
+    SpellInfo const* _spellInfoHeal;
+    SpellEffectInfo const* _healEffectDummy;
+};
+
+// 33076 - Prayer of Mending
+class spell_pri_prayer_of_mending : public SpellScriptLoader
+{
+public:
+    spell_pri_prayer_of_mending() : SpellScriptLoader("spell_pri_prayer_of_mending") { }
+
+    class spell_pri_prayer_of_mending_SpellScript : public spell_pri_prayer_of_mending_SpellScriptBase
+    {
+        PrepareSpellScript(spell_pri_prayer_of_mending_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_AURA });
+        }
+
+        void HandleEffectDummy(SpellEffIndex effIndex)
+        {
+            SpellEffectInfo const* effectInfo = GetEffectInfo(effIndex);
+            if (!effectInfo)
+                return;
+
+            Unit* target = GetExplTargetUnit();
+            if (!target)
+                return;
+
+            Unit* caster = GetCaster();
+            CastPrayerOfMendingAura(caster, target, effectInfo->BasePoints);
```

Comment:
Use `GetEffectValue()`, not `effectInfo->BasePoints`, that gives you the value including all possible modifiers (as a general rule, you should never need to access raw effectInfo->BasePoints in any script)